src/services/group.service.ts

```diff
@@ -0,0 +1,110 @@
+import {Groups, sequelize, Users} from '../loaders/database';
+import {v4} from 'uuid';
+import {RESPONSE_MESSAGES} from '../config/messages';
+import {
+  GROUPS_RETURN_ATTRIBUTES,
+  USERS_RETURN_ATTRIBUTES,
+} from '../config/return-params';
+import {Permissions} from '../config/enums';
+
+type Permission = Record<Permissions, string>[];
+
+const INCLUDE_USERS = {
+  model: Users,
+  attributes: USERS_RETURN_ATTRIBUTES,
+  through: {
+    attributes: [],
+  },
+};
+
+export default {
+  async getGroups() {
+    return await Groups.findAll({
+      include: INCLUDE_USERS,
+      attributes: GROUPS_RETURN_ATTRIBUTES,
+    });
+  },
+
+  async getGroup(id: string) {
+    const group = await Groups.findByPk(id, {
+      include: INCLUDE_USERS,
+      attributes: GROUPS_RETURN_ATTRIBUTES,
+    });
+    if (!group) {
+      throw new Error(RESPONSE_MESSAGES.GROUP_NOT_FOUND);
+    }
+    return group;
+  },
+
+  async createGroup(name: string, permissions: Permission) {
```

Comment:
permissions is array of string

```suggestion
  async createGroup(name: string, permissions: Permissions[]) {
```